lambdas/update_supporter_appointment_settings.py

```diff
@@ -13,42 +13,70 @@ def update_supporter_appointment_settings(event, context):
     job_search = event['job_search']
     grad_student = event['grad_student']
 
-    #Connect to table
-    client = boto3.client('rds-data')
+    preferences = {}
+    specializations = {}
+    majors = {}
 
     #Check if supporter exists
-    existing_user = client.execute_statement(
-        secretArn = constants.SECRET_ARN, 
-        database = constants.DB_NAME,
-        resourceArn = constants.ARN,
-        sql = "SELECT supporter_id FROM supporters WHERE supporter_id = '%d';" % (supporter_id)
-    )
-    if(existing_user['records'] == []):
-        return{
-            'statuscode' : 404
+    sql = 'SELECT * FROM supporters WHERE supporter_id = :supporter_id;'
+    sql_parameters = [{'name': 'supporter_id', 'value': {'longValue': supporter_id}}]
+    response = query(sql, sql_parameters)
+    if(response['records'] == []):
+        return {
+            'body': json.dumps("The supporter does not exist"),
+            'statusCode': 404
         }
 
-    query = ""
-
-    #Can't be null
+    #Adding to preferences dictionary
+    #The reason I add to the dictionary by adding another dictionary with the value type
+    #and then value is for the parameterization
+    if(job_search != None):
+        preferences['job_search'] = job_search
+    if(grad_student != None):
+        preferences['grad_student'] = grad_student
     if(job_search == None or grad_student == None): #These values cannot be null
```

Comment:
Consider checking for these values separately and returning an error message in the body of the response. This way, we can more easily debug and the front end can diagnose the problem.